### DISCUSSION TITLE
Detect blobs

### DISCUSSION THREAD

**Discussion by lbdc** [Breakthrough/DVR-Scan#164] [Ideas]
Just wondering if there is a way for DVR-Scan to detect "blobs of pixel" or "continuous areas" to prevent false detection due to rain, snow and noise.

If not could this be implemented as enhancement?

**Comment by Breakthrough** [comment]
It's definitely possible, one could supply an image as a mask (e.g. a transparent .png).  Or were you thinking of something more automated, like blurring the image and using a morphological filter on the more stable areas of the image?

In the meantime, note that if you get the latest source code, you can supply a rectangular ROI ([see this comment](https://github.com/Breakthrough/DVR-Scan/issues/3#issuecomment-507718860) for details).

**Comment by Breakthrough** [comment]
Hopefully with adjusting kernel size and the new region editor, it should be easier to mask out certain areas and small blobs.

**Comment by mmokrejs** [comment]
I don't think what the original poster meant. If the infrared shines into water droplet or falling snowflake while the shutter speed is rather slow then such blobs (of merely white pixels) move vertically (or in all kinds of direction if the wind is deflected from the wall). Masking a region does not help and one cannot increase kernel size to get rid of these false motion hits.

The algorithm would first take into account it is moving from top to the bottom to catch most of the situations and second, that there are typically many such trajectories, they have white color provided it was captured during night. is the current time accessible in the stream? Or information it is captured with IR-cut filter enabled?

Needless to say a heavy rain during the day triggers also same false alarm.

The motion algorith should better work with a (BW?) sketch of the image and ignore this type of "noise" in front. At has to work with more abstract glimpse of the scene.

**Comment by elvis-epx** [comment]
One thing that could help is a "max-area" threshold. Thinking about trying my hand myself. The rationale is, if the bounding box is too big, it is a sympthom of rain/haze (see screenshot for an example).

<img width="1624" height="1056" alt="Screenshot 2025-07-10 at 21 17 14" src="https://github.com/user-attachments/assets/e81a2387-60d5-4088-86fc-d616afb2b8a3" />

**Reply by Breakthrough** [reply]
This is an excellent example.  To test this theory, could you experiment by [changing the max-threshold option](https://github.com/Breakthrough/DVR-Scan/blob/main/dvr-scan.cfg#L98-L100)?  It looks like from your screen grab the score there is ~95, so setting it a bit lower to like 80 might be a good test.  That should make those frames get counted as having no motion.  This will cause some real motion events to be skipped potentially, so we do need to come up with a fix in the source code regardless.

Edit: If you are willing to provide some sample footage for the project and unit tests, that would be super helpful!  Just a 30 second clip would be fine.

**Reply by elvis-epx** [reply]
Drafted some work at https://github.com/Breakthrough/DVR-Scan/pull/224
